test/e2e/index.ts

```diff
@@ -1441,6 +1441,52 @@ test(
   })
 );
 
+test(
+  "client reconnects websocket if closed before making a zome call",
+  withConductor(ADMIN_PORT, async (t) => {
+    const { cell_id, client, admin } = await installAppAndDna(ADMIN_PORT, true);
+    await admin.authorizeSigningCredentials(cell_id);
+    await client.client.close();
+    const call = client.callZome({
+      cell_id,
+      zome_name: TEST_ZOME_NAME,
+      fn_name: "bar",
+      provenance: cell_id[1],
+      payload: null,
+    });
+    try {
+      await call;
+      t.pass("websocket was reconnected successfully");
+    } catch (error) {
+      t.fail("websocket was not reconnected");
+    }
+  })
+);
+
+test(
+  "client fails to reconnect to websocket if closed before making a zome call and does not end up in a loop",
```

Comment:
This isn't quite testing what I was concerned about with looping. It's repeatedly reconnecting with an invalid token that I was concerned about. If the caller isn't careful about which error is which then they might treat this error as retryable and keep trying to reconnect. I'd prefer if this test could make two zome calls and the second one got a different error message back asking the caller to provide new credentials

Reply:
Well, configuring the message to as for new credentials doesn't work I think due to the "client closed with pending requests" error that doesn't even let us define the error message here.

Reply:
Maybe I can look into handling this close event better in the first place.

Reply:
But I'm not sure whether we should conflate this with this PR

Reply:
There is no way of modifying the error in a way to ask the user to provide a valid token. Any zome call with an invalid token will now lead to the following error:
```
ClientClosedWithPendingRequests: client closed with pending requests - close event code: 1005, request id: 1
```
And this seems to be a result of how the conductor handles things and cannot be affected at the js side as I understand it. This error however is distinguishable by a caller from a "normal" zome call error if required so I don't think there's more we can do here without touching the conductor behavior.